### PR TITLE
WIP: fix for memory error in qdel

### PR DESCRIPTION
**Pull request by bremanandjk**
#### Describe Bug or Feature

Memory leaks and invalid read/write errors.

#### Describe Your Change

Instead of freeing an array of strings, free() was used earlier. That would be potential a cause for memory leaks and errors. 
So, changed the code to a free array of strings.

### CODE REVIEW

**Comment by prakashcv13** [comment]
@bremanandjk, could you please post the log of reported invalid read/writes reported by valgrind, both before and after your fix.

File: src/server/process_request.c [hunk 1]
```diff
@@ -1517,7 +1517,7 @@ free_br(struct batch_request *preq)
 			break;
 		case PBS_BATCH_DeleteJobList:
 			if (preq->rq_ind.rq_deletejoblist.rq_jobslist)
-				free(preq->rq_ind.rq_deletejoblist.rq_jobslist);
+				free_string_array(preq->rq_ind.rq_deletejoblist.rq_jobslist);
```
Comment (bhroam) [inline src/server/process_request.c:1520]: The difference between free() and free_string_array() will be a leak, not invalid read/writes.  How does this fix the invalid read/writes?

Comment (bremanandjk) [reply src/server/process_request.c:1520]: I have tried to rerun the test which found the invalid read/write and memory leaks, seeks memory issues were not reported in valgrind. Let me try multiple times and confirm the test results. Meanwhile, let mark PR as WIP.

**Comment by bremanandjk** [comment]
> @bremanandjk, could you please post the log of reported invalid read/writes reported by valgrind, both before and after your fix.
I ran the same test by the running server under Valgrind, I couldn't get any memory-related errors specific to qdel request. 
logs: [valgrind.log](https://github.com/openpbs/openpbs/files/5639260/valgrind.log)

**Comment by agrawalravi90** [comment]
I saw a server crash which seems to also be related to the same codepath, it happened when i submitted a few hundred jobs to the server with scheduling on and then did qdel `qselect`. I think it has something to do with the fact that some jobs ran and finished from the list that the client sent to the server, can you please take a look at this as well?

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000148db656b877 in __strlen_avx2 () from /usr/lib64/libc.so.6
[Current thread is 1 (Thread 0x148db8f9f500 (LWP 39728))]
Missing separate debuginfos, use: yum debuginfo-install cyrus-sasl-lib-2.1.27-5.el8.x86_64 expat-2.2.5-4.el8.x86_64 glibc-2.28-127.el8.x86_64 keyutils-libs-1.5.10-6.el8.x86_64 krb5-libs-1.18.2-5.el8.x86_64 libblkid-2.32.1-24.el8.x86_64 libcom_err-1.45.6-1.el8.x86_64 libgcc-8.3.1-5.1.el8.x86_64 libical-3.0.3-3.el8.x86_64 libicu-60.3-2.el8_1.x86_64 libmount-2.32.1-24.el8.x86_64 libpq-12.4-1.el8_2.x86_64 libselinux-2.9-4.el8_3.x86_64 libstdc++-8.3.1-5.1.el8.x86_64 libuuid-2.32.1-24.el8.x86_64 libxcrypt-4.1.1-4.el8.x86_64 openldap-2.4.46-15.el8.x86_64 openssl-libs-1.1.1g-11.el8.x86_64 python3-libs-3.6.8-31.el8.x86_64 systemd-libs-239-41.el8_3.x86_64 zlib-1.2.11-16.el8_2.x86_64
(gdb) bt
#0  0x0000148db656b877 in __strlen_avx2 () from /usr/lib64/libc.so.6
#1  0x0000148db646052f in vfprintf () from /usr/lib64/libc.so.6
#2  0x0000148db6487c24 in vsnprintf () from /usr/lib64/libc.so.6
#3  0x0000148db64673d3 in snprintf () from /usr/lib64/libc.so.6
#4  0x0000000000469229 in req_deletejob (preq=0x1708e00) at ../../../src/server/req_delete.c:426
#5  0x000000000046b4f0 in post_delete_route (pwt=0x1706ac0) at ../../../src/server/req_delete.c:1371
#6  0x00000000004eb47a in dispatch_task ()
#7  0x00000000004eb7d5 in default_next_task ()
#8  0x0000000000464134 in next_task () at ../../../src/server/pbsd_main.c:1611
#9  0x0000000000463ad0 in main (argc=1, argv=0x7ffe9570ecc8) at ../../../src/server/pbsd_main.c:1344
(gdb) up
#1  0x0000148db646052f in vfprintf () from /usr/lib64/libc.so.6
(gdb) up
#2  0x0000148db6487c24 in vsnprintf () from /usr/lib64/libc.so.6
(gdb) up
#3  0x0000148db64673d3 in snprintf () from /usr/lib64/libc.so.6
(gdb) up
#4  0x0000000000469229 in req_deletejob (preq=0x1708e00) at ../../../src/server/req_delete.c:426
426			snprintf(jid, sizeof(jid), "%s", jobids[j]);
(gdb) p jobids
$1 = (char **) 0x17073d0
(gdb) p jobids[j]
$2 = 0x148db67c0038 <error: Cannot access memory at address 0x148db67c0038>
(gdb) p j
$3 = 145
(gdb) p jobids[144]
$4 = 0x637362702e333034 <error: Cannot access memory at address 0x637362702e333034>
(gdb) p jobids[143]
$5 = 0x21 <error: Cannot access memory at address 0x21>
(gdb) p jobids[142]
$6 = 0x3b23 <error: Cannot access memory at address 0x3b23>
(gdb) p jobids[141]
$7 = 0x1707850 "403.pbsc8"
```

**Comment by bremanandjk** [comment]
Raised another PR along with this change.

**Pull request closed without merging**